src/Compilers/CSharp/Portable/Symbols/NullableFlowStateExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal static class NullableFlowStateExtensions
+    {
+        internal static CodeAnalysis.NullableFlowState ToPublicFlowState(this NullableFlowState nullableFlowState)
+        {
+            Debug.Assert((CodeAnalysis.NullableFlowState)(NullableFlowState.NotNull + 1) == CodeAnalysis.NullableFlowState.NotNull);
+            return (CodeAnalysis.NullableFlowState)nullableFlowState + 1;
```

Comment:
>(CodeAnalysis.NullableFlowState)nullableFlowState + 1; [](start = 19, length = 54)

Consider an explicit switch instead

Reply:
I preferred to do it this way: it's significantly more efficient as there's no branching involved.

---
In reply to: [265766419](https://github.com/dotnet/roslyn/pull/34017#discussion_r265766419) [](ancestors = 265766419)